plugins/woocommerce/src/Admin/API/Reports/Orders/DataStore.php

```diff
@@ -623,12 +627,36 @@ public static function get_all_statuses() {
 			);
 			/* phpcs:enable */
 
-			Cache::set( $cache_key, $statuses );
+			set_transient( 'woocommerce_analytics_orders_statuses_all', $statuses, YEAR_IN_SECONDS );
 		}
 
 		return $statuses;
 	}
 
+	/**
+	 * Ensure the order status will present in `get_all_statuses` call result.
+	 *
+	 * @param int $order_id Order ID.
+	 */
+	public static function actualize_all_statuses_transient_if_necessary( $order_id ) {
+		$order = wc_get_order( $order_id );
+		if ( $order ) {
+			$status = self::normalize_order_status( $order->get_status() );
+			if ( 'shop_order_refund' === $order->get_type() ) {
+				$parent_order = wc_get_order( $order->get_parent_id() );
```

Comment:
Do we need to get the parent here?  I wonder if can skip this section since any updates to the parent will also be processed separately.

Reply:
I replicated the original logic from here: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/API/Reports/Orders/Stats/DataStore.php#L510. Didn't dig into the original reasoning though.

Reply:
I believe that's just needed for the lookup tables to store a status around the refund.  We can probably forgo that logic here since the original order would have already run through here.

Reply:
Ok, I'll simplify - based on the built context, we should be good.

Reply:
Addressed in da98c43c6bf56b430b8fa24e899c03e5f619ca88